documentation/modules/oauth/proc-oauth-authentication-broker-config.adoc

```diff
@@ -88,11 +88,33 @@ external:
     clientSecret: <3>
       secretName: my-cluster-oauth
       key: clientSecret
+    userNameClaim: preferred_username <4>
 ----
 --
 <1> URI of the token introspection endpoint.
 <2> Client ID to identify the client.
 <3> Client Secret and client ID is used for authentication.
+<4> The token claim (or key) that contains the actual user name in the token. The user name is the _principal_ used to identify the user. The `userNameClaim` value will depend on the authorization server used.
++
+Depending on how you apply {oauth} authentication, and the type of authorization server, there are additional (optional) configuration settings you can use:
++
+[source,yaml,subs="+quotes,attributes"]
+----
+  # ...
+  authentication:
+    type: oauth
+    # ...
+    checkIssuer: false <1>
+    fallbackUserNameClaim: client_id <2>
+    fallbackUserNamePrefix: client-account- <3>
+    validTokenType: bearer <4>
+    userInfoEndpointUri: <__https://<auth-server-address>/auth/realms/external/protocol/openid-connect/userinfo__> <5>
+----
+<1> Some authorization servers may not provide `iss` claim making it impossible to perform an issuer check. In that case do not specify `validIssuerUri` and set `checkIssuer` to `false`. Default is `true`.
+<2> Some authorization servers may not provide a single attribute that can be used to identify both regular users, and clients. If client authenticates in its own name `client_id` may be provided, but if user is authenticated using username and password to obtain a refresh token or an access token, some `username` attribute may be provided in addition to `client_id`. This config option specifies the claim (attribute) to use id primary user id attribute is not available.
+<3> In situations where `fallbackUserNameClaim` is applicable it may also be necessary to prevent name collisions between the values of username claim, and those of fallback username claim. Consider a situaction where a client called `producer` exists, but also a regular user called `producer` exists. We must be able to differenciate between the two, which we do by prefixing the user id for client.
```

Comment:
```suggestion
<3> In situations where `fallbackUserNameClaim` is applicable, it may also be necessary to prevent name collisions between the values of the username claim, and those of the fallback username claim. Consider a situation where a client called `producer` exists, but also a regular user called `producer` exists. In order to differentiate between the two, you can use this property to add a prefix to the user id of the client.
```